vignettes/articles/startup_messages.Rmd

```diff
@@ -0,0 +1,149 @@
+                       force = TRUE, upgrade = "never")
+
+```
+
+Now, we can load the newly created package.     
+
+```{r}
+# Detach packages
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+
+# Load library
+library(startuppackage)
+```
+
+Upon loading, {startuppackage} produces several warnings and messages that we would like to silence:   
+- a message informing that {dplyr}, a dependency of the {startuppackage}, is required      
+- a warning about the conflicts in {dplyr}     
+- a startup message specific to {startuppackage}      
+- a warning about the conflicts in {startuppackage}       
+
+
+## Loading a library without warnings
+
+To suppress the warnings about conflicts, you can set the `warn.conflict` argument to `FALSE` in `library()`. As written in the documentation of the `library()` function, a conflict is a function masking a function, or a non-function masking a non-function.
+
+```{r}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+library(startuppackage, warn.conflicts = FALSE)
+```
+  
+Note that this removes only the warnings regarding the conflicts of the {startuppackage}, not the conflicts relative to the {dplyr} package, which is a dependency from the {startuppackage}.
+
+
+## Loading a library without messages
+
+To suppress the messages regarding the attachement of a package (either confirming the attachement of the package or warning about failed attachement), set the `quietly` argument in the `library()` function to `TRUE`. This will suppress messages relative to dependencies, but not warnings about conflicts.
+
+```{r}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+library(startuppackage, quietly = TRUE)
+```
+
+Note that here, it only removed the message regarding the existence of the {dplyr} dependency.             
+
+
+## Loading a library silently with the function suppressPackageStartupMessages()
+
+`suppressPackageStartupMessages()` ignores diagnostic messages when loading a package. It also ignores warning messages about conflicts.
+```{r}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+suppressPackageStartupMessages(library(startuppackage))
+```
+
+Here the `suppressPackageStartupMessages()` function suppressed all messages and warnings. 
+
+
+## Loading a library silently within a chunk
+
+If the code is written in R markdown, another way to silence the warnings and messages is to use the chunk options. Make a separate chunk for loading the librairies and set the chunk options `message` and `warning` to `FALSE`. Write the rest of the reprex in a second chunk to prevent hiding useful warnings and messages for debugging.
+
+```{r, message=FALSE, warning=FALSE}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+library(startuppackage)
+```
+
+
+## Loading the tidyverse library silently
```

Comment:
This section needs to reflect that `reprex::repres()` already exposes this via the argument `tidyverse_quiet`. This defaults to `TRUE` but can be toggled in any individual call and can also be set in a user's `.Rprofile`. No one using reprex should ever be setting this option themselves "by hand".